File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    student = [[], [1,2,3,4,5], [2,1,2,3,2,4,2,5], [3,3,1,1,2,2,4,4,5,5]]
    cnt = [0]*4
    
    for i in range(1, len(student)):
        j = 0
        k = 0
        while k < len(answers):
            t = answers[k]
            if t == student[i][j]:
                cnt[i] += 1
            j += 1
            k += 1
            if j == len(student[i]):
                j = 0
    _max = max(cnt)
    for idx, data in enumerate(cnt):
        if data == _max:
            answer.append(idx)
    return answer

File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public int[] solution(int[] answers) {
        int[] score = new int[3];
        List<Integer> answer = new ArrayList<>();
        int[][] patterns = {{1, 2, 3, 4, 5}, {2, 1, 2, 3, 2, 4, 2, 5}, {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}};
        
        for(int i=0; i<score.length; i++){
            int count = 0;
            for(int j=0; j<answers.length; j++){
                if(answers[j] == patterns[i][j % patterns[i].length])
                    count++;
            }
            score[i] = count;
        }
        
        int maxCount = Math.max(score[0], Math.max(score[1], score[2]));
        
        for(int i=0; i<score.length; i++){
            if(maxCount == score[i])
                answer.add(i+1);
        }
        
        return answer.stream().mapToInt(value -> value).toArray();
    }
}
